idaes/apps/uncertainty_propagation/tests/test_uncertainties.py

```diff
@@ -0,0 +1,252 @@
+# Engineering Framework (IDAES PSE Framework) Copyright (c) 2018-2020, by the
+# software owners: The Regents of the University of California, through
+# Lawrence Berkeley National Laboratory,  National Technology & Engineering
+# Solutions of Sandia, LLC, Carnegie Mellon University, West Virginia
+# University Research Corporation, et al. All rights reserved.
+#
+# Please see the files COPYRIGHT.txt and LICENSE.txt for full copyright and
+# license information, respectively. Both files are also available online
+# at the URL "https://github.com/IDAES/idaes-pse".
+##############################################################################
+import sys
+import os
+sys.path.append(os.path.abspath('..')) # current folder is ~/tests
+import numpy as np
+import pandas as pd
+import pytest
+from pytest import approx
+from mock import patch
+from idaes.apps.uncertainty_propagation.uncertainties import quantify_propagate_uncertainty, propagate_uncertainty, get_sensitivity, clean_variable_name
+from pyomo.opt import SolverFactory
+from pyomo.environ import *
+import pyomo.contrib.parmest.parmest as parmest
+
+ipopt_available = SolverFactory('ipopt').available()
+kaug_available = SolverFactory('k_aug').available()
+dotsens_available = SolverFactory('dot_sens').available()
+
+class TestUncertaintyPropagation:
+
+    @pytest.mark.unit
+    @pytest.mark.skipif(not ipopt_available, reason="The 'ipopt' command is not available")
+    @pytest.mark.skipif(not ipopt_available, reason="The 'k_aug' command is not available")
+    @pytest.mark.skipif(not ipopt_available, reason="The 'dot_sens' command is not available")
+    def test_quantify_propagate_uncertainty1(self):
+        from idaes.apps.uncertainty_propagation.examples.rooney_biegler import rooney_biegler_model,rooney_biegler_model_opt
+        variable_name = ['asymptote', 'rate_constant']
+        data = pd.DataFrame(data=[[1,8.3],[2,10.3],[3,19.0],
+                                  [4,16.0],[5,15.6],[7,19.8]],
+                            columns=['hour', 'y'])
+        def SSE(model, data):
+            expr = sum((data.y[i] - model.response_function[data.hour[i]])**2 for i in data.index)
+            return expr
+    
+        obj, theta, cov, propagation_f, propagation_c =  quantify_propagate_uncertainty(rooney_biegler_model,rooney_biegler_model_opt, data, variable_name, SSE)
+
+        assert obj == approx(4.331711213656886)
+        assert theta[variable_name[0]] == approx(19.142575284617866)
+        assert theta[variable_name[1]] == approx(0.53109137696521)
+        assert cov == approx(np.array([[6.30579403, -0.4395341], [-0.4395341, 0.04193591]]))
+        assert propagation_f['objective'] == approx(5.45439337747349)
+        assert propagation_c == {}
+        
+        
+    def test_quantify_propagate_uncertainty2(self):
+        from idaes.apps.uncertainty_propagation.examples.rooney_biegler import rooney_biegler_model
```

Comment:
Please add a docstring here that examples in 1 or 2 sentences the goal of the test. This will be really helpful in a few months if someone wants to make changes.

Reply:
Added